test/e2e/certificates_test.go

```diff
@@ -0,0 +1,478 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/cert-manager-operator/test/library"
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+
+	v1 "github.com/cert-manager/cert-manager/pkg/apis/acme/v1"
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	certmanagermetav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ACME Certificate", Ordered, func() {
+	t := &testing.T{}
+
+	var ctx context.Context
+	var appsDomain string
+
+	BeforeAll(func() {
+		By("creating Kube clients")
+		ctx = context.Background()
+		loader = library.NewDynamicResourceLoader(ctx, t)
+
+		config, err := library.GetConfigForTest(t)
+		Expect(err).NotTo(HaveOccurred())
+
+		configClient, err = configv1.NewForConfig(config)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
Thanks for pointing out, I did find an inconsistency here. The `Context("ACME Certificate")` has an init for `loader` while `Context("Self-signed Certificate")` doesn't, but loader is used by both. This can quickly get confusing and invite nil pointer refs in the future. 

I'll rectify and move these initialization in the `BeforeSuite` irrespective of their usage in specific Context(s) which should help naturally keep the init and the var declaration together atleast.

Reply:
Updated!